// Copyright 2018 The Hypebot Authors. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package hypebot.riot.v3;

import "hypebot/protos/riot/v3/constants.proto";

service LeagueService {
  rpc ListLeaguePositions(ListLeaguePositionsRequest)
      returns (ListLeaguePositionsResponse) {
  }
}

message ListLeaguePositionsRequest {
  int64 summoner_id = 1;
}

message ListLeaguePositionsResponse {
  repeated LeaguePosition positions = 1;
}

enum TierRank {
  INVALID_RANK = 0;

  I = 1;
  II = 2;
  III = 3;
  IV = 4;
  V = 5;
}

message LeaguePosition {
  TierRank rank = 1;
  QueueType queue_type = 2;
  bool hot_streak = 3;
  MiniSeries mini_series = 4;
  int32 wins = 5;
  bool veteran = 6;
  int32 losses = 7;
  string player_or_team_id = 8;
  string league_name = 9;
  string player_or_team_name = 10;
  bool inactive = 11;
  bool fresh_blood = 12;
  Tier tier = 13;
  int32 league_points = 14;
}

message MiniSeries {
  int32 wins = 1;
  int32 losses = 2;
  int32 target = 3;
  string progress = 4;
}
